src/server/pkg/storage/fileset/transforms.go

```diff
@@ -0,0 +1,95 @@
+package fileset
+
+import (
+	"io"
+
+	"github.com/pachyderm/pachyderm/src/server/pkg/storage/fileset/index"
+	"github.com/pachyderm/pachyderm/src/server/pkg/storage/fileset/tar"
+)
+
+var _ FileSource = &HeaderFilter{}
+
+type HeaderFilter struct {
+	F func(th *tar.Header) bool
```

Comment:
I think I prefer descriptive field names. The one letter variable names should be limited to variables that have a small scope.